### DISCUSSION TITLE
Error: Cannot find package *** imported from ***

### DISCUSSION THREAD

**Discussion by tech-daedalus** [vercel/community#6003] [Help]
### Summary

Having issues with using Vercel with Typescript, ESM, and Typescript Paths

* package.json has `type: "module"`
* tsconfig.json has ` "@/*": ["./*"]` as its path
* tsconfig.json has `"module": "NodeNext",`
* All imports have the `.js` extension needed for ESM

### Steps to Reproduce

The Typescript path works perfectly in `middleware.ts` but any file within `api` cannot seem to import files from `utils` which is on the same directory level

![Screenshot 2024-03-04 at 08 49 42](https://github.com/vercel/community/assets/157588555/9479d5f1-9b18-42b3-9fe1-201b7422c51c)

## Error
![Screenshot 2024-03-04 at 08 51 15](https://github.com/vercel/community/assets/157588555/0ab7a3a1-ef26-4511-b5df-32c97ab9a2e3)

**Comment by devotox** [comment]
Similar Issue here - https://github.com/orgs/vercel/discussions/1226

**Reply by tech-daedalus** [reply]
Everything works from there. Only difference is the Typescript path
